Lib/test/test_sys.py

```diff
@@ -812,6 +812,22 @@ def __del__(self):
         rc, stdout, stderr = assert_python_ok('-c', code)
         self.assertEqual(stdout.rstrip(), b'True')
 
+    @test.support.requires_type_collecting
+    def test_issue20602(self):
+        # sys.flags and sys.float_info were wiped during shutdown.
+        code = """if 1:
+            import sys
+            class A:
+                def __del__(self):
+                    print(sys.flags)
```

Comment:
There are no guaranties that the reference to `a` will be cleared before clearing the reference to `sys`. Keep a local reference:
```python
def __del__(self, sys=sys):
```